arduino-ide-extension/src/browser/dialogs/ide-updater/ide-updater-component.tsx

```diff
@@ -7,36 +6,32 @@ import ReactMarkdown from 'react-markdown';
 import { ProgressInfo, UpdateInfo } from '../../../common/protocol/ide-updater';
 import ProgressBar from '../../components/ProgressBar';
 
-export type IDEUpdaterComponentProps = {
-  updateInfo: UpdateInfo;
-  windowService: WindowService;
+export type UpdateProgress = {
+  progressInfo?: ProgressInfo | undefined;
   downloadFinished?: boolean;
   downloadStarted?: boolean;
-  progress?: ProgressInfo;
   error?: Error;
-  onDownload: () => void;
-  onClose: () => void;
-  onSkipVersion: () => void;
-  onCloseAndInstall: () => void;
+};
+
+export type IDEUpdaterComponentProps = {
```

Comment:
[minor]: Why did you decide to use `type` instead of `interface` here?

Reply:
It's just habit, but for me, it's pretty much the same. Since across this project we typically use interfaces, I'll happily change it as you request, but can I ask you why you prefer it?

Reply:
> Interfaces are generally preferred over type literals because interfaces can be implemented, extended and merged.

From [here](https://github.com/bradzacher/eslint-plugin-typescript/blob/master/docs/rules/prefer-interface.md).